src/modules/Dashboard/features/Applet/Schedule/ActivityForm/ActivityForm.tsx

```diff
@@ -1,67 +1,189 @@
+    {
+      setRemoveAllScheduledPopupVisible,
+      setRemoveAlwaysAvailablePopupVisible,
+      submitCallback,
+      setActivityName,
+      defaultStartDate,
+    },
     ref,
   ) => {
+    const [selectValues, setSelectValues] = useState<null | Option[]>(null);
     const { t } = useTranslation('app');
+    const appletData = applets.useAppletData();
+    const appletId = appletData?.result.id;
 
+    //TODO: add filling up the form with the data of the edited event when the get events API is connected
     const methods = useForm<FormValues>({
-      defaultValues,
+      resolver: yupResolver(ActivityFormSchema()),
+      defaultValues: getDefaultValues(defaultStartDate),
       mode: 'onChange',
     });
 
     const {
-      getValues,
       handleSubmit,
       control,
       formState: { dirtyFields },
+      watch,
     } = methods;
 
-    const submitForm = () => {
-      if (dirtyFields.availability) {
-        if (getValues().availability) {
-          setRemoveAllEventsPopupVisible(true);
+    const activityOrFlowId = watch('activityOrFlowId');
+    const alwaysAvailable = watch('alwaysAvailable');
+    const oneTimeCompletion = watch('oneTimeCompletion');
+    const timerType = watch('timerType');
+    const timerDuration = watch('timerDuration');
+    const idleTime = watch('idleTime');
+    const periodicity = watch('periodicity');
+    const startTime = watch('startTime');
+    const endTime = watch('endTime');
+    const accessBeforeSchedule = watch('accessBeforeSchedule');
+    const date = watch('date');
+    const startEndingDate = watch('startEndingDate');
+
+    // TODO: add get events callback
+    const { execute, error } = useAsync(createEventApi);
+
+    const handleCreateEvent = async () => {
+      if (!appletId) {
+        return;
+      }
+
+      const regexWithFlow = /^flow-/;
+      const isFlowId = regexWithFlow.test(activityOrFlowId);
+
+      const body: CreateEventType['body'] = {
+        timerType,
+        ...(isFlowId
+          ? { flowId: activityOrFlowId.replace(regexWithFlow, '') }
+          : { activityId: activityOrFlowId }),
+      };
+
+      if (timerType === TimerType.Timer) {
+        body.timer = addSecondsToHourMinutes(timerDuration);
+      }
+
+      if (timerType === TimerType.Idle) {
+        body.timer = addSecondsToHourMinutes(idleTime);
+      }
+
+      if (alwaysAvailable) {
+        body.oneTimeCompletion = oneTimeCompletion;
+        body.periodicity = {
+          type: 'ALWAYS',
+          selectedDate: format(defaultStartDate as Date, DateFormats.YearMonthDay),
+        };
+      } else {
+        body.startTime = addSecondsToHourMinutes(startTime);
+        body.endTime = addSecondsToHourMinutes(endTime);
+        body.accessBeforeSchedule = accessBeforeSchedule;
+
+        body.periodicity = {
+          type: periodicity,
+          ...(periodicity === Periodicity.Once
+            ? {
+                selectedDate: convertDateToYearMonthDay(date as Date),
```

Comment:
is there no way to fix type not to do everywhere as Date

Reply:
done